lib/routes/account.js

```diff
@@ -385,7 +386,17 @@ module.exports = function (
               function (result) {
                 emailRecord = result
 
-                doSigninConfirmation = requestHelper.shouldEnableSigninConfirmation(emailRecord, config, request)
+                // Session token verification is only enabled for certain users during phased rollout.
+                // Even when it is enabled, we only do the email challenge if:
+                //  * the request wants keys, since unverified sessions are fine to use for e.g. oauth login.
+                //  * the email is verified, since content-server triggers a resend of the verification
+                //    email on unverified accounts, which doubles as sign-in confirmation.
+                if (! requestHelper.shouldEnableTokenVerification(emailRecord, config, request)) {
+                  tokenVerificationId = undefined
+                  doSigninConfirmation = false
+                } else {
+                  doSigninConfirmation = requestHelper.wantsKeys(request) && emailRecord.emailVerified
```

Comment:
I'm hazy, and this may require some explaining, possibly in the code. I believe the security this feature provides comes from the existence of the `tokenVerificationId` even if keys are not requested. The existence of `tokenVerificationId` for a given sessionToken means the session is not considered verified and that it cannot be used to fetch keys from `/account/keys`?